src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -41,4 +43,18 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
+
+    @Override
+    public SizeResponse restoreSize(UUID sizeId){
+        Size size = findSizeByIdOrThrowException(sizeId);
+        validateSizeForRestoration(size);
+        size.setDeleted(false);
+        return SizeMapper.toSizeResponse(sizeRepository.save(size));
+    }
+
+    public void validateSizeForRestoration(Size size) {
+        if(!size.isDeleted()){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + size.getSizeId() + " is not deleted." );
+        }
+    }
```

Comment:
El nombre del método debería ser más representativo a lo que hace, en este caso solo estamos viendo si está eliminado, por lo que podría llamarse de otra manera

```suggestion
    public void ifSizeIsAlreadyDeletedThrowException(Size size) {
        if(!size.isDeleted()){
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + size.getSizeId() + " is not deleted." );
        }
    }
```